paimon-core/src/main/java/org/apache/paimon/catalog/CachingCatalog.java

```diff
@@ -111,7 +119,15 @@ public CachingCatalog(
                         .expireAfterAccess(expirationInterval)
                         .ticker(ticker)
                         .build();
+        this.partitionCache =
+                Caffeine.newBuilder()
+                        .softValues()
+                        .executor(Runnable::run)
+                        .expireAfterAccess(expirationInterval)
+                        .ticker(ticker)
```

Comment:
Can you use `maximumWeight(cachedPartitionMaxNum)`?
You can also set `weigher`.